api/v1alpha1/component_types.go

```diff
@@ -38,18 +39,11 @@ type GitSource struct {
 	DockerfileURL string `json:"dockerfileUrl,omitempty"`
 }
 
-// ComponentSource describes the Component source
-type ComponentSource struct {
-	ComponentSourceUnion `json:",inline"`
-}
-
-type ComponentSourceUnion struct {
-	// Git Source for a Component
-	GitSource *GitSource `json:"git,omitempty"`
-}
+// Component is the Schema for the components API. This is used to generate any k8s deployment resource
```

Comment:
I would generally tend to avoid discussing about any API that this library is not aware of, considering this is to be a standalone library

Reply:
Changing it to:
```
// Component - An Application contains one or more components. This captures information about a component's
// deployment, service and route.
```

Although, the name of the struct, `Component` could also be changed.   eg. GeneratorOptions